homeassistant/components/dsmr_reader/sensor.py

```diff
@@ -0,0 +1,87 @@
+"""Support for DSMR Reader through MQTT."""
+import logging
+
+from homeassistant.components import mqtt
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import slugify
+
+from .definitions import DEFINITIONS
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "dsmr_reader"
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up DSMR Reader sensors."""
+
+    sensors = []
+    for topic in DEFINITIONS:
+        sensors.append(DSMRSensor(topic))
+
+    async_add_entities(sensors)
+
+
+class DSMRSensor(Entity):
+    """Representation of a DSMR sensor that is updated via MQTT."""
+
+    def __init__(self, topic):
+        """Initialize the sensor."""
+
+        self._definition = DEFINITIONS[topic]
+
+        self._entity_id = slugify(topic.replace("/", "_"))
+        self._topic = topic
+
+        self._name = self._definition["name"]
+        self._unit_of_measurement = (
+            self._definition["unit"] if "unit" in self._definition else ""
```

Comment:
We can replace the ternary operator with `dict.get`. It's built for this case.
```py
self._unit_of_measurement = self._definition.get("unit")
```

Reply:
We shouldn't use empty string for unit of measurement. If there's no unit, return `None` as unit of measurement.